totaluptime/provider.go

```diff
@@ -0,0 +1,337 @@
+	if err != nil {
+		return nil, err
+	}
+
+	var providerRecords TotalUptimeRecords
+	err = p.performAPIcall(req, &providerRecords)
+	if err != nil {
+		return nil, err
+	}
+
+	// A records
+	for _, val := range providerRecords.ARecord.Rows {
+		ttl, err := strconv.Atoi(val.ATTL)
+		if err != nil {
+			return nil, err
+		}
+
+		record := libdns.Record{
+			ID:    val.ID,
+			Type:  "A",
+			Name:  val.AHostName,
+			Value: val.AIPAddress,
+			TTL:   time.Duration(ttl) * time.Second,
+		}
+		records = append(records, record)
+	}
+
+	// CNAME records
+	for _, val := range providerRecords.CNAMERecord.Rows {
+		ttl, err := strconv.Atoi(val.CnameTTL)
+		if err != nil {
+			return nil, err
+		}
+
+		record := libdns.Record{
+			ID:    val.ID,
+			Type:  "CNAME",
+			Name:  val.CnameName,
+			Value: val.CnameAliasFor,
+			TTL:   time.Duration(ttl) * time.Second,
+		}
+		records = append(records, record)
+	}
+
+	// MX records
+	for _, val := range providerRecords.MXRecord.Rows {
+		ttl, err := strconv.Atoi(val.MxTTL)
+		if err != nil {
+			return nil, err
+		}
+
+		record := libdns.Record{
+			ID:    val.ID,
+			Type:  "MX",
+			Name:  val.MxDomainName,
+			Value: val.MxMailServer,
+			TTL:   time.Duration(ttl) * time.Second,
+		}
+		records = append(records, record)
+	}
+
+	// NS records
+	for _, val := range providerRecords.NSRecord.Rows {
+		ttl, err := strconv.Atoi(val.NsTTL)
+		if err != nil {
+			return nil, err
+		}
+
+		record := libdns.Record{
+			ID:    val.ID,
+			Type:  "NS",
+			Name:  val.NsHostName,
+			Value: val.NsName,
+			TTL:   time.Duration(ttl) * time.Second,
+		}
+		records = append(records, record)
+	}
+
+	// SRV records (research needed)
+	// 	// TODO: possible compiler or static-check error (Go v1.19.3):
+	// 	// issue 1) claims Weight is not part of this struct
+	// 	// issue 2) claims Priority is an int (should be uint)
+
+	// TXT records
+	for _, val := range providerRecords.TXTRecord.Rows {
+		ttl, err := strconv.Atoi(val.TxtTTL)
+		if err != nil {
+			return nil, err
+		}
+
+		record := libdns.Record{
+			ID:    val.ID,
+			Type:  "TXT",
+			Name:  val.TxtHostName,
+			Value: val.TxtText,
+			TTL:   time.Duration(ttl) * time.Second,
+		}
+		records = append(records, record)
+	}
+
+	return records, nil
+}
+
+// AppendRecords adds records to the zone. It returns the records that were added.
+func (p *Provider) AppendRecords(ctx context.Context, zone string, records []libdns.Record) ([]libdns.Record, error) {
+	domain := getDomain(zone)
+	domainID := p.getDomainID(domain)
+	var successAppended []libdns.Record
+
+	if domainID == "" {
+		return nil, fmt.Errorf("record append cannot proceed: unknown zone: %s", zone)
+	}
+
+	// append each record individually
+	for _, rec := range records {
+		// identify record type
+		typeName, err := convertRecordTypeToProvider(rec.Type)
+		if err != nil {
+			log.Printf("Unable to identify record type; skipping: %s: %s\n", rec.Type, rec.Name)
+			continue // skip to next record
```

Comment:
```suggestion
			return nil, err
```

There's a lot of lines like this... ideally, don't emit logs -- just do, or do not. (There is no try.)

I'm not familiar with TU, so it's up to you decide: if this case is unexpected, then return the error. If it's normal and harmless (doesn't affect the final result; i.e. this is a debug-level log), then remove the log and just continue.